### PR TITLE
[luci/pass] Extend ForwardReshapeToUnaryOpPass with Mul and Div ops

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This commit extends ForwardReshapeToUnaryOpPass with Mul and Div ops.

for issue: https://github.com/Samsung/ONE/issues/11956
from draft https://github.com/Samsung/ONE/pull/12124

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

**Review by jinevening — DISMISSED** [review]
LGTM

File: compiler/luci/pass/src/ForwardReshapeToUnaryOpPass.cpp [hunk 1]
```diff
@@ -201,6 +298,8 @@ namespace luci
  *          |            |           |
  *
  *   UnaryOp: CircleNeg, ...
+ *   Note: Binary Op (Div, Mul) can also be considered as a unary operation
+ *   if one of its inputs is a constant.
```
Comment (seanshpark) [inline compiler/luci/pass/src/ForwardReshapeToUnaryOpPass.cpp:302]:
```suggestion
 *   Note: Binary Op (Div, Mul) can also be considered as a unary operation
 *         if one of its inputs is a constant.
```

Comment (BalyshevArtem) [reply compiler/luci/pass/src/ForwardReshapeToUnaryOpPass.cpp]: Done

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by jinevening — APPROVED** [review]

**Pull request merged**